rule/add_constant.go

```diff
@@ -36,13 +38,17 @@ type AddConstantRule struct {
 	allowList       allowList
 	ignoreFunctions []*regexp.Regexp
 	strLitLimit     int
-
-	configureOnce sync.Once
+	configureErr    error
+	configureOnce   sync.Once
 }
 
 // Apply applies the rule to given file.
-func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
-	r.configureOnce.Do(func() { r.configure(arguments) })
+func (r *AddConstantRule) Apply(file *lint.File, arguments lint.Arguments) ([]lint.Failure, error) {
+	r.configureOnce.Do(func() { r.configureErr = r.configure(arguments) })
+
+	if r.configureErr != nil {
+		return nil, r.configureErr
+	}
```

Comment:
I would use this


to avoid adding a variable to a struct that you only use in the scope of one method


```suggestion
	var err error
	r.configureOnce.Do(func() { err = r.configure(arguments) })

	if err != nil {
		return nil, err
	}
```

But this would fail only on first run

So maybe what you are looking for is to use a sync.OnceFunc and not a syncOne

This way you can do this


```suggestion
	err := r.configureOnce.Do(func() error { return r.configure(arguments) })

	if err != nil {
		return nil, err
	}
```

Reply:
@ccoVeille i choose first suggestion, because second one generates errors:

```
r.configureOnce.Do((func() error literal)) (no value) used as value compiler (TooManyValues)                                                                
cannot use (func() error literal) (value of type func() error) as func() value in argument to r.configureOnce.Do compiler (IncompatibleAssign) 
```